pkg/server/backend_manager.go

```diff
@@ -225,6 +227,7 @@ func (s *DefaultBackendStorage) RemoveBackend(identifier string, idType pkgagent
 	}
 	if len(s.backends[identifier]) == 0 {
 		delete(s.backends, identifier)
+		metrics.Metrics.BackendDec()
```

Comment:
This seems correct but brittle. I might be better to call Set(len(s.backends)) rather than calling Inc and Dec and hoping you didn't miss a use case.